mixins/localize-mixin.js

```diff
@@ -136,24 +136,25 @@ export const LocalizeMixin = dedupeMixin(superclass => class LocalizeMixinClass
 
 	}
 
-	localizeHTML(key, { _links = '', _html = '', _tooltips = '', ...replacements } = {}) {
-		const links = [].concat(_links);
-		const tooltips = [].concat(_tooltips);
-		const html = [].concat(_html);
+	localizeHTML(key, {
+		_link = '', _tooltip = '', _html = '',
+		_links = {}, _tooltips = {},
+		...replacements
+	} = {}) {
 		const localized = this.localize(key, replacements);
 		const sanitized = localized.replace(/</g, '&lt;').replace(/>/g, '&gt;');
 		const markedUp = sanitized
-			// replace [a], [a 0] ... [a 9], [/a]
-			.replace(/\[a( [0-9])?\]([^]*?)\[\/a\]/g, (m, n, t) => `<d2l-link ${links[Number(n || 0)] || ''}>${t}</d2l-link>`)
-			// replace [tt], [tt 1] ... [tt 9], [/tt]
-			.replace(/\[tt( [0-9])?\]([^]*?)\[\/tt\]/g, (m, n, t) => `<d2l-tooltip-help inherit-font-style text="${t}">${tooltips[Number(n || 0)] || ''}</d2l-tooltip-help>`)
-			// UNDOCUMENTED: replace [html], [html 0] ... [html 9]
-			.replace(/\[html( [0-9])?\]/g, (m, n) => html[Number(n || 0)] || '')
+			// replace [link], [link key], [/link]
+			.replace(/\[a(?: ?([\w-]*))?\]([^]*?)\[\/a\]/g, (m, n, t) => `<d2l-link ${_links[n] || _link}>${t}</d2l-link>`)
+			// replace [tooltip-help], [tooltip-help key], [/tooltip-help]
+			.replace(/\[tooltip-help(?: ?([\w-]*))?\]([^]*?)\[\/tooltip-help\]/g, (m, n, t) => `<d2l-tooltip-help inherit-font-style text="${t}">${_tooltips[n] || _tooltip}</d2l-tooltip-help>`)
```

Comment:
Maybe `href`s only need to be HTML-encoded... 🤔

Reply:
The text from the lang file (i.e. `${t}`) definitely needs to replace double quotes so that `[tooltip-help]text" onmouseover="import('malicious.js'[/tooltip-help]` doesn't work. And only because it gets moved to the attribute value.

The rest of the values are passed from the code or have `<` and `>` replaced, making them unable to produce parseable html, so shouldn't need to be modified further unless I'm totally missing something.